pkg/engine/validate/pattern.go

```diff
@@ -183,6 +183,9 @@ func validateValueWithStringPattern(log logr.Logger, value interface{}, pattern
 
 	operator := operator.GetOperatorFromStringPattern(pattern)
 	pattern = pattern[len(operator):]
+	// remove leading and trailing white spaces before
+	// getting numerical and string parts
+	pattern = strings.TrimSpace(pattern)
```

Comment:
Minor feedback: Is this comment necessary? A general rule that I find good to follow is that if the comment is (re)stating what is easily read in code we should remove the comment.

Reply:
Will keep that in mind, thanks! I was unsure if this was needed or not in the first place 😅